bin/recursive_gen/CMakeLists.txt

```diff
@@ -0,0 +1,143 @@
+#---------------------------------------------------------------------------#
+# Copyright (c) 2018-2022 Mikhail Komarov <nemo@nil.foundation>
+# Copyright (c) 2020-2022 Nikita Kaskov <nbering@nil.foundation>
+#
+# Distributed under the Boost Software License, Version 1.0
+# See accompanying file LICENSE_1_0.txt or copy at
+# http://www.boost.org/LICENSE_1_0.txt
+#---------------------------------------------------------------------------#
+
+cmake_minimum_required(VERSION 3.5)
+
+cm_find_package(CM)
+include(CMDeploy)
+include(CMSetupVersion)
+
+cm_project(recursive_gen WORKSPACE_NAME ${CMAKE_WORKSPACE_NAME} LANGUAGES ASM C CXX)
+
+if(NOT CMAKE_CROSSCOMPILING)
+    find_package(Boost COMPONENTS REQUIRED filesystem log log_setup program_options thread system)
+elseif(CMAKE_CROSSCOMPILING)
+    if(NOT TARGET boost)
+        include(ExternalProject)
+        set(Boost_LIBRARIES boost_random)
+        externalproject_add(boost
+                            PREFIX ${CMAKE_BINARY_DIR}/libs/boost
+                            GIT_REPOSITORY git@github.com:boostorg/boost.git
+                            GIT_TAG boost-1.77.0
+                            BUILD_IN_SOURCE TRUE
+                            CMAKE_ARGS -DCMAKE_CROSSCOMPILING_EMULATOR=${CMAKE_CROSSCOMPILING_EMULATOR} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+                            BUILD_COMMAND cmake --build . --target ${Boost_LIBRARIES}
+                            INSTALL_COMMAND "")
```

Comment:
I don't think anyone has ever tried to cross-compile any of our code. 
In addition, this could lead to inconsistent builds and hidden replacement of installed Boost package